File: src/cli/format.ts
import type { CommandResult, ExitCode } from './types.js';

/** Builds a successful CommandResult, choosing JSON or human text output. */
export function ok(json: boolean, jsonPayload: unknown, text: string): CommandResult {
  return { exitCode: 0, stdout: json ? `${JSON.stringify(jsonPayload)}\n` : text };
}

/** Builds a failing CommandResult (exit code 1 or 2), choosing JSON or human text output. */
export function fail(
  exitCode: Exclude<ExitCode, 0>,
  json: boolean,
  jsonPayload: unknown,
  text: string,
): CommandResult {
  if (json) {
    return { exitCode, stdout: `${JSON.stringify(jsonPayload)}\n` };
  }
  return { exitCode, stdout: '', stderr: text };
}

File: python/src/masterytrace/cli/format.py
"""Builds successful/failing CommandResults, choosing JSON or human text output. Ported from src/cli/format.ts."""
from __future__ import annotations

import json
from typing import Any

from .types import CommandResult


def ok(as_json: bool, json_payload: Any, text: str) -> CommandResult:
    return CommandResult(exit_code=0, stdout=f"{json.dumps(json_payload)}\n" if as_json else text)


def fail(exit_code: int, as_json: bool, json_payload: Any, text: str) -> CommandResult:
    if as_json:
        return CommandResult(exit_code=exit_code, stdout=f"{json.dumps(json_payload)}\n")
    return CommandResult(exit_code=exit_code, stdout="", stderr=text)
